src/components/SelectionList/BaseSelectionList.tsx

```diff
@@ -1026,10 +1009,10 @@ function BaseSelectionList<TItem extends ListItem>({
                         ListFooterComponent={
                             <>
                                 {footerContentAbovePagination}
-                                {listFooterContent ?? ShowMoreButtonInstance}
+                                {listFooterContent}
                             </>
                         }
-                        onEndReached={onEndReached}
+                        onEndReached={onEndReached ?? incrementPage}
```

Comment:
Inside the `incrementPage` function, we can use condition `flattenedSections.allOptions.length > CONST.MAX_SELECTION_LIST_PAGE_LENGTH * currentPage;` to increase the page value. Currently, it is increasing irrespective of there is more data or not on scrolling the end of the list.